content/components/banner.mdx

```diff
@@ -5,18 +5,293 @@ componentId: banner
+
+  </div>
+</Box>
+
+### Title
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="banner without title and banner with title"
+    src="https://github.com/primer/primitives/assets/813754/c5cc31d6-dafc-426d-999b-215f68ca7c28"
+  />
+
+Use a title to concisely present the topic of the banner when the message of the banner is complex and too long to scan. 
+
+A title is required and will be used as the accessibility title. However, you can visually hide the title. It is recommended to hide the title if the body of the banner is short and easily scannable.
+
+</Box>
+
+### Dismissable
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="banners with a dismiss button"
+    src="https://github.com/primer/primitives/assets/813754/85c94192-eb1a-4e7c-8fc0-b3c5f31ac251"
+  />
+  <div>
+
+#### Dismiss button
+
+Banners with an `onDismiss` property automatically show a dismiss button. This should be used for banners that inform about something that the user can not solve. For example issues with the organization they are part of or informational banners. When it makes sense, the choice to dismiss the banenr should be stored and the banner should not be shown again.
+
+  </div>
+</Box>
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="a critical banner with no actions and one with a single action"
+    src="https://github.com/primer/primitives/assets/813754/a8684eed-5544-4d6d-a1c6-eda1376e95b5"
+  />
+  <div>
+
+#### Exception: Critical banner
+
+Critical banners can't be dismissable. They are mostly used for errors which need to be resolved. As a result you can only get rid of a critical banner by resolving the error. If used to warn about   destructive actions, the banner should also not be dismissable.
+
+If the error must be resolved on a different page, use an action that directs the user to the page where the error can be resolved.
+
+  </div>
+</Box>
+
+### Actions
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="trailing actions on a banner"
+    src="https://github.com/primer/primitives/assets/813754/7c57bc61-ee0d-4673-86e9-550b945b117a"
+  />
+  <div>
+
+A maximum of 2 actions can be added to a banner. Whenever possible it is recommended to only add a single action to better direct the users attention.
+
+For banners with a single action, you can choose between adding a primary or secondary action depending on how much attention you want to draw to the action.
```

Comment:
I see what you are saying. However, we have a default button and a invisible button here. The default has a lot more visual prominence.

I am happy to update names (please suggest something that works better), but I think it is important to keep a reference to one being more prominent and if used together this one should be the "suggest" action, hence primary in the current naming.

We don't know the context, sometimes a single button action may be very important, sometimes it may be less so, which is why consumer can choose 
the type.